mutablesecurity/solutions/implementations/duplicati/code.py

```diff
@@ -0,0 +1,693 @@
+                DuplicatiHelper._make_local_backup(file_test, backup_path)
+                + f" && dir {backup_path} |grep '*.f{EncryptionModule.get()}'",
+                f"rm -r {backup_path}",
+            ]
+
+            return execute_command
+
+        @staticmethod
+        def process(output: typing.List[str]) -> bool:
+            return int(output[0]) != 0
+
+    IDENTIFIER = "encryption_test"
+    DESCRIPTION = "Checks if the Duplici local encryption works"
+    TEST_TYPE = TestType.SECURITY
+    FACT = EncryptionTest
+
+
+# Solution class definition
+
+
+class Duplicati(BaseSolution):
+    INFORMATION = [
+        BinaryArchitecture,
+        CompressionModule,
+        EncryptionModule,
+        ExcludeFilesAttributes,
+        SkipFilesLarger,
+        RestorePath,
+        Passphrase,
+    ]
+    TESTS = [
+        ClientEncryption,
+        SupportedArchitecture,
+        ClientCommandPresence,
+    ]
+    LOGS = [
+        DefaultLogs,  # type: ignore[list-item]
+    ]
+    ACTIONS = [
+        LocalBackup,  # type: ignore[list-item]
+        RestoreLocalBackup,  # type: ignore[list-item]
+        SshBackup,  # type: ignore[list-item]
+        RestoreSshBackup,  # type: ignore[list-item]
+        GoogleDriveBackup,  # type: ignore[list-item]
+        RestoreGoogleDriveBackup,  # type: ignore[list-item]
+    ]
+
+    @staticmethod
+    @deploy
+    def _install() -> None:
+        architecture = BinaryArchitecture.get()
+        if not architecture:
+            raise IncompatibleArchitectureException()
+
+        release_url = (
+            "https://updates.duplicati.com/beta/duplicati_2.0.6.3-1_all.deb"
+        )
+        apt.deb(
+            name="Install Duplicati via deb",
+            src=release_url,
+        )
+
+        template_path = os.path.join(
+            os.path.dirname(__file__), "files/duplicati.conf.j2"
+        )
+        j2_values = {
+            "encryption_module": EncryptionModule.get(),
+            "compression_module": CompressionModule.get(),
+            "skip_files_larger_than": SkipFilesLarger.get(),
+            "exclude_files_attributes": ExcludeFilesAttributes.get(),
+            "passphrase": Passphrase.get(),
+        }
+        files.template(
+            src=template_path,
+            dest="/opt/mutablesecurity/duplicati/duplicati.conf",
+            configuration=j2_values,
+            name="Copy the generated configuration into Duplicati's folder.",
+        )
+
+    @staticmethod
+    @deploy
+    def _uninstall(remove_logs: bool = True) -> None:
+        apt.packages(
+            packages=["duplicati"],
+            present=False,
+            extra_uninstall_args="--purge",
+            name="Uninstalls Duplicati via apt.",
+        )
+        files.directory(
+            name="Remove duplicati executable and configuration.",
+            path="/opt/mutablesecurity/duplicati",
+            present=False,
+        )
+        files.directory(
```

Comment:
As `/var/log/duplicati.log` is a file, please use `files.file` instead of `files.directory`.